pkg/cache/cache.go

```diff
@@ -264,8 +264,7 @@ func (c *vSphereCache) GetDatastoreMoByReference(ctx context.Context, dcName str
 			return *cds.dsMo, nil
 		}
 	}
-	err = fmt.Errorf("couldn't find Datastore with reference %+v in datacenter %s", ref, dcName)
-	klog.Error(err)
+	klog.V(4).Infof("couldn't find Datastore with reference %+v in datacenter %s (this is expected if datastore is not related to OCP)", ref, dcName)
 	return mo.Datastore{}, err
```

Comment:
This now returns `nil` instead of an error. Is that expected?

Reply:
Actually...it would be better if we do return error here and let the caller just log info - changed.